### PR TITLE
fix: console showing subworkflows as unknown

### PR DESCRIPTION
**Pull request by olga-union**
Signed-off-by: Olga Nad <olga@union.ai>

# TL;DR
Some subworkflows compared non-empty objects with `null`, which resulted in the failure to extract fields on a `null` and the whole stack of unresolved computational steps - adding an early exit out of the comparison function, if one of the parameters is null/undefined, resolves the issue with improper populated data in the node executions table.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Demo
Before: 
Screen Shot 2022-08-19 at 10 33 49 AM

After:
Screen Shot 2022-08-19 at 10 34 17 AM

## Tracking Issue
fixes https://github.com/flyteorg/flyteconsole/issues/406

## Follow-up issue
_NA_

### CODE REVIEW

File: packages/zapp/console/src/components/Executions/Tables/nodeExecutionColumns.tsx [hunk 1]
```diff
@@ -107,15 +106,6 @@ const DisplayType: React.FC<NodeExecutionCellRendererData> = ({ execution }) =>
...
 
-const hiddenCacheStatuses = [CatalogCacheStatus.CACHE_MISS, CatalogCacheStatus.CACHE_DISABLED];
-function hasCacheStatus(taskNodeMetadata?: TaskNodeMetadata): taskNodeMetadata is TaskNodeMetadata {
-  if (!taskNodeMetadata) {
-    return false;
-  }
-  const { cacheStatus } = taskNodeMetadata;
-  return !hiddenCacheStatuses.includes(cacheStatus);
-}
-
```
Comment (olga-union) [inline packages/zapp/console/src/components/Executions/Tables/nodeExecutionColumns.tsx:118]: removed unused code

File: packages/zapp/console/src/components/WorkflowGraph/utils.ts [hunk 2]
```diff
@@ -35,6 +35,11 @@ export function isExpanded(node: any) {
  * @returns     boolean
  */
 export const checkIfObjectsAreSame = (a, b) => {
+  // if one of the objects is null (undefined), objects can't be the same
+  if ((!a || !b) && a != b) {
```
Comment (ursucarina) [inline packages/zapp/console/src/components/WorkflowGraph/utils.ts:39]: nit: if these are objects, then the equality check might not be enough.

How deep does the comparison need to be? would `lodash.isEqual(a, b) `be a better alternative?

**Review by ursucarina — APPROVED** [review]

File: packages/zapp/console/src/components/WorkflowGraph/utils.ts (hunk 2 above)
Comment (olga-union) [reply packages/zapp/console/src/components/WorkflowGraph/utils.ts:39]: That's a good idea - we can completely replace this helper with `_.isEqual`, I was working on top of existing code and thought about using `_.has` which doesn't help with `null` values, but replacing the whole this with isEqual is even better!

**Review by ursucarina — APPROVED** [review]

**Review by jsonporter — APPROVED** [review]
lgtm
